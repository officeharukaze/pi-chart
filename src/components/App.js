import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { PieChart, Pie } from 'recharts';
import '../styles/App.css';

const style = {
  xxx: {
    margin: 0,
    padding: 0,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  yyy: {
    position: 'fixed',
  },
  button: {
    border: '1px solid #999',
    borderRadius: 10,
    width: 60,
    fontSize: '1.2em',
    fontWeight: 'bold',
    cursor: 'pointer',
    background: '#ccc',
    margin: 5,
  },
  wrapper: {
    display: 'flex',
    justifyContent: 'center',
  },
};

export const PureApp = ({count, data, dispatch}) => {
  const color = data[0].value === 1 ? '#0f9' : '#f00';
  return (
    <div className="App">
      <div style={style.xxx}>
        <PieChart width={200} height={200}>
          <Pie
            dataKey="value"
            data={data}
            startAngle={90}
            endAngle={450}
            cx={95}
            cy={95}
            innerRadius={40}
            outerRadius={80}
            fill={color}
            animationEasing="ease-in"
          />
        </PieChart>
        <div style={style.yyy}>{count}ä»¶</div>
      </div>
      <div style={style.wrapper}>
        <div style={style.button} onClick={
          () => {
            dispatch({ type: 'INIT' });
            setTimeout(() => dispatch({ type: 'INC' }), 5);
          }
        }>+</div>
        <div style={style.button} onClick={
          () => {
            dispatch({ type: 'INIT' });
            setTimeout(() => dispatch({ type: 'DEC' }), 5);
          }
        }>-</div>
      </div>
    </div>
  )
};

PureApp.propTypes = {
  count: PropTypes.number,
  data: PropTypes.array,
  dispatch: PropTypes.func,
};

const mapStateToProps = state => ({
  count: state.main.count,
  data: state.main.data,
});

export default connect(mapStateToProps)(PureApp);
