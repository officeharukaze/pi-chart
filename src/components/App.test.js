import React from 'react';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';
import { PureApp as App } from './App';

it('renders without crashing', () => {
  const component = renderer.create(
    <App count={1} />
  );
  expect(component).toMatchSnapshot();
});
