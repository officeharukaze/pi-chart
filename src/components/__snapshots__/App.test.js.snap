// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Welcome to React
    </h2>
  </div>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.jsx
    </code>
     and save to reload.
  </p>
  <button
    onClick={[Function]}
  >
    INC
  </button>
  1
  <button
    onClick={[Function]}
  >
    DEC
  </button>
</div>
`;
